### ISSUE TITLE
git stash

### ISSUE THREAD

**Issue by bmll-chris** [jupyterlab/jupyterlab-git#309]
Similarly to the `History` tab, it would be useful to have a tab for managing stashed changes.

**Comment by FMelloMascarenhas-Cohere** [comment]
+1

**Comment by fcollonval** [comment]
cc @shawnesquivel 

Let's start simple we could add two commands:
- `Stash` to stash all changes (except untracked files)
- `Pop latest stash`

## On the backend

you will need to create two methods in the `Git` class:
- one that execute `git stash`
- one that execute `git stash pop`

> You can find more about the [Git commands](https://git-scm.com/docs/git-stash) in the documentation.

The you will need to add a handler for those in `handlers.py`:

```py
class GitStashHandler(GitHandler):
    @tornado.web.authenticated
    async def post(self, path: str = ""):
        # ... call git stash
   
    @tornado.web.authenticated
    async def delete(self, path: str = "", name: str = ""):
        # ... call git stash pop
```

You can look for example at 

https://github.com/jupyterlab/jupyterlab-git/blob/35f712f3e745f77ae6316bcfa11670c2ad5f20ed/jupyterlab_git/handlers.py#L398-L409

and 

https://github.com/jupyterlab/jupyterlab-git/blob/35f712f3e745f77ae6316bcfa11670c2ad5f20ed/jupyterlab_git/handlers.py#L430-L440

Then you will need to bind a new endpoint with the handlers at

https://github.com/jupyterlab/jupyterlab-git/blob/35f712f3e745f77ae6316bcfa11670c2ad5f20ed/jupyterlab_git/handlers.py#L901

I advice you to add some tests to check your implementation - see e.g. https://github.com/jupyterlab/jupyterlab-git/blob/master/jupyterlab_git/tests/test_remote.py

## On the frontend

You need to add two methods for those two actions in 

https://github.com/jupyterlab/jupyterlab-git/blob/35f712f3e745f77ae6316bcfa11670c2ad5f20ed/src/model.ts#L35

Then you will need to add two commands that call those methods; taking inspiration from

https://github.com/jupyterlab/jupyterlab-git/blob/35f712f3e745f77ae6316bcfa11670c2ad5f20ed/src/commandsAndMenu.tsx#L199-L243

To test them, you could add them to the menu at

https://github.com/jupyterlab/jupyterlab-git/blob/master/src/commandsAndMenu.tsx#L1320

I let you imagine a UI for those features.

**Comment by fcollonval** [comment]
Closing as resolved y #1228
